components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/DefaultIdentityProviderImpl.java

```diff
@@ -108,6 +108,36 @@ public String getIdOfUser(String userName) throws IdentityProviderException {
     }
 
     @Override
+    public String getEmailOfUser(String userId) throws IdentityProviderException {
+      Response userResponse = scimServiceStub.getUser(userId);
+        String userEmail;
+        if (userResponse == null) {
+            String errorMessage =
+                    "Error occurred while retrieving Id of user " + userId + ". Error : Response is null.";
+            throw new IdentityProviderException(errorMessage, ExceptionCodes.RESOURCE_RETRIEVAL_FAILED);
+        }
+        if (userResponse.status() == APIMgtConstants.HTTPStatusCodes.SC_200_OK) {
+            String responseBody = userResponse.body().toString();
+            JsonParser parser = new JsonParser();
+            JsonObject parsedResponseBody = (JsonObject) parser.parse(responseBody);
+            userEmail =  parsedResponseBody.get("emails").toString().replaceAll("[\\[\\]\"]", "");
+
+
+            String message = "Email " + userEmail + " of user " + parsedResponseBody.get(USERNAME).getAsString()
+                    + " is successfully retrieved from SCIM endpoint.";
+            if (log.isDebugEnabled()) {
+                log.debug(message);
+            }
+        } else {
+            String errorMessage =
```

Comment:
Do we log this exception anywhere in our flow?

Reply:
Yes, Fixed with PR: https://github.com/wso2/carbon-apimgt/pull/4537